pkg/cluster/sync_test.go

```diff
@@ -63,44 +63,47 @@ func noEmptySync(cluster *Cluster, err error, reason SyncReason) error {
 
 func TestConnectionPoolerSynchronization(t *testing.T) {
 	testName := "Test connection pooler synchronization"
-	var cluster = New(
-		Config{
-			OpConfig: config.Config{
-				ProtectedRoles: []string{"admin"},
-				Auth: config.Auth{
-					SuperUsername:       superUserName,
-					ReplicationUsername: replicationUserName,
-				},
-				ConnectionPooler: config.ConnectionPooler{
-					ConnectionPoolerDefaultCPURequest:    "100m",
-					ConnectionPoolerDefaultCPULimit:      "100m",
-					ConnectionPoolerDefaultMemoryRequest: "100Mi",
-					ConnectionPoolerDefaultMemoryLimit:   "100Mi",
-					NumberOfInstances:                    int32ToPointer(1),
+	newCluster := func() *Cluster {
```

Comment:
Why these changes are necessary for the `CloneDescription`?

Reply:
Ping, should this be removed for the scope of having a smaller PR=

Reply:
@Jan-M do you want to remove test completely?

Reply:
@erthalion these changes fix error during manual creation of `postgresqls.acid.zalan.do` CRD.

Reply:
> @erthalion these changes fix error during manual creation of postgresqls.acid.zalan.do CRD.

That's the goal of the whole PR, isn't it? I'm just trying to figure out if there is any mysterious value in changing unit test that seems to be completely unrelated (connection pooler test).

Reply:
That unit test was broken with new functionality:
`--- FAIL: TestConnectionPoolerSynchronization (0.00s)`

Objects for testing before changes were static objects, after change introduced they became pointers... So before changes, the test just copies static objects and takes the address of it. And now copying pointer will still point to the same object! So tests always ruined the same underlying object with mock copies.

That's why all new mocked objects now created using `New()`, but not copying.

I hope it's clear enough :)

Reply:
But only `CloneDescription` became a point, right? And it's not involved in this test, so still sounds strange for me. But ok, at least it's not included by mistake.

Reply:
The thing is the test itself was written for static properties. As soon as we've added pointers the simple copying became a problem. That's why we do not copy mock objects but create them separately so they do not interfere